src/uct/ib/rdmacm/rdmacm_iface.c

```diff
@@ -202,8 +203,10 @@ static void uct_rdmacm_client_handle_failure(uct_rdmacm_iface_t *iface,
 {
     ucs_assert(!iface->is_server);
     if (ep != NULL) {
+        pthread_mutex_lock(&ep->mutex);
         uct_rdmacm_ep_set_failed(&iface->super.super, &ep->super.super, status);
         uct_rdmacm_ep_invoke_completions(ep, status);
+        pthread_mutex_unlock(&ep->mutex);
```

Comment:
is it safe to call user callback with lock held? could there be a deadlock here?
better let uct_rdmacm_ep_invoke_completions() take the lock and drop it every time before calling a completion